server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -513,7 +513,14 @@ class FileSorter {
       }
     }
 
-    public void sort() {
+    public void sort() throws InterruptedException {
+      int reserveMemory = (int) initialReserveSingleSortMemory;
+      memoryTracker.reserveSortMemory(reserveMemory);
+      while (!memoryTracker.sortMemoryReady()) {
+        Thread.sleep(20);
+      }
```

Comment:
Do we need to roll back this change(put the reserve memory and check memory status inside the sort() )? Before I change it, this method seems would only be called by single thread `fileSorterSchedulerThread`, but now it would be called by the mutil-threads of `fileSorterExecutors`, and we may need to add a lock to this method ensure thread-safe,  and this lock will be competed when each time calling sort(). I would like to confirm whether we need to keep this change which may require locking? @waitinfuture @AngersZhuuuu

Reply:
I think we'd better to rollback this change.